src/aws_iot_mqtt_client_common_internal.c

```diff
@@ -631,13 +631,10 @@ IoT_Error_t aws_iot_mqtt_internal_wait_for_read(AWS_IoT_Client *pClient, uint8_t
 			break;
 		}
 		rc = aws_iot_mqtt_internal_cycle_read(pClient, pTimer, &read_packet_type);
-	} while(NETWORK_DISCONNECTED_ERROR != rc && read_packet_type != packetType);
+	} while((SUCCESS == rc || MQTT_NOTHING_TO_READ == rc) && read_packet_type != packetType);
```

Comment:
Sorry for being picky but I would rather have:
while(((SUCCESS == rc) || (MQTT_NOTHING_TO_READ == rc)) && (read_packet_type != packetType));

Reply:
Fixed.